File: LeetCode Problems/101 to 200/121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int l = 0, mx = 0, r = 1;
        while (r < prices.size()) {
            if (prices[r] < prices[l])
                l = r;
            else
                mx = max(mx, prices[r] - prices[l]);
            r += 1;
        }
        return mx;
    }
};

File: LeetCode Problems/101 to 200/121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mx = 0, l = 0, r = 1;
        while (r < prices.length) {
            if (prices[r] <= prices[l])
                l = r;
            else
                mx = Math.max(mx, prices[r] - prices[l]);
            r++;
        }
        return mx;
    }
}
